crates/shielded_token/src/vp.rs

```diff
@@ -624,15 +624,33 @@ where
                 return Err(error);
             }
         }
+        let masp_tx_id = shielded_tx.txid().into();
+        let has_trans_inputs = shielded_tx
+            .transparent_bundle()
+            .map(|b| !b.vin.is_empty())
+            .unwrap_or(false);
         // The transaction shall not push masp authorizer actions that are not
-        // needed cause this might lead vps to run a wrong validation logic
-        if !actions_authorizers.is_empty() {
+        // needed because this might lead vps to run a wrong validation logic
+        if !actions_authorizers.is_empty() && !has_trans_inputs {
             let error = Error::new_const(
                 "Found masp authorizer actions that are not required",
             );
             tracing::debug!("{error}");
             return Err(error);
         }
+        // check that the tx has a shielding fee section
```

Comment:
Don't we also need to check that the right amount of fees has been paid with a supported token and to the expected recipient?